### PR TITLE
Added Euclidean GCD [Golang]

### LINKED ISSUE #342: Euclidean GCD [Golang]
#### This issue is about [Euclidean GCD](https://en.wikipedia.org/wiki/Euclidean_algorithm).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by rashinanwani**
fixes #342

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: euclidean_gcd/euclidean_gcd.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	if b != 0 {
+		return EuclideanGCDRecursive(b, (a % b))
+	}
+	return a
+}
+
+func main() {
+	a := 90
+	b := 65
+	fmt.Println("GCD of nos is :")
```
Comment (aashutoshrathi) [inline euclidean_gcd/euclidean_gcd.go]: change `nos` to `numbers`

**Comment by yashLadha** [comment]
Please Resolve the conflicts associated with the `README`

File: euclidean_gcd/euclidean_gcd.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package main
+
+import "fmt"
+
+// EuclideanGCDIterative finds GCD of given numbers by iterative method
```
Comment (yashLadha) [inline euclidean_gcd/euclidean_gcd.go]: It would be great if you explain the information about the parameters and show the time complexity of the function as well.

File: euclidean_gcd/euclidean_gcd.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+func EuclideanGCDIterative(a int, b int) int {
+	for b != 0 {
+		tmp := b
+		b = a % b
+		a = tmp
+	}
+	return a
+}
+
+// EuclideanGCDRecursive finds GCD of given numbers by recursion
```
Comment (yashLadha) [inline euclidean_gcd/euclidean_gcd.go]: Please show the Recurrence equation of the algorithm, and also give the time complexity as well

File: euclidean_gcd/euclidean_gcd.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+func EuclideanGCDRecursive(a int, b int) int {
+	if b != 0 {
+		return EuclideanGCDRecursive(b, (a % b))
+	}
+	return a
+}
+
+func main() {
+	a := 90
+	b := 65
```
Comment (yashLadha) [inline euclidean_gcd/euclidean_gcd.go]: Try to give better name to variables. Instead of `a` and `b` , you can use `firstNumber` or `secondNumber` as well.

File: euclidean_gcd/euclidean_gcd.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		second = first % second
+		first = tmp
+	}
+	return first
+}
+
+// EuclideanGCDRecursive finds GCD of given numbers by recursion
+// first : first number
+// second : second number
+// Recurrance relation : If second=0 GCD is first number otherwise fuction is called recursively to determine GCD
```
Comment (yashLadha) [inline euclidean_gcd/euclidean_gcd.go]: Show the recurrence equation of the function.

**Comment by yashLadha** [comment]
Conflicting `README.md` please see through it, and try to resolve the conflicts

**Comment by yashLadha** [comment]
Can you rebase the minor commits. I would look nice. :smiley:

**Comment by yashLadha** [comment]
Resolve the README conflicts

File: README.md [hunk 6]
```diff
@@ -17,18 +17,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
 | [Binary Search](https://en.wikipedia.org/wiki/Binary_search_algorithm) | [:white_check_mark:](binary_search/binary_search.c) | | [:white_check_mark:](binary_search/BinarySearch.java) | [:white_check_mark:](binary_search/binary_search.py) | [:white_check_mark:](binary_search/binary_search.go) | [:white_check_mark:](binary_search/binarySearch.js) |
-| [Coin Change Problem](http://www.algorithmist.com/index.php/Coin_Change) | [:white_check_mark:](coin_change_problem/coin_change_problem.c) | | [:white_check_mark:](coin_change_problem/CoinChange.java) | [:white_check_mark:](coin_change_problem/coin_change_problem.py) | | |
-| [Counting Sort](http://www.geeksforgeeks.org/counting-sort/)| [:white_check_mark:](counting_sort/counting_sort.c) | | [:white_check_mark:](counting_sort/CountingSort.java) | | | |
+| [Coin Change Problem](http://www.algorithmist.com/index.php/Coin_Change) | [:white_check_mark:](coin_change_problem/coin_change_problem.c) | | [:white_check_mark:](coin_change_problem/CoinChange.java) | [:white_check_mark:](coin_change_problem/coin_change_problem.py) | | [:white_check_mark:](coin_change_problem/coinChangeProblem.js) |
```
Comment (yashLadha) [inline README.md]: Don't add changes that are not related to this PR.

File: euclidean_gcd/euclidean_gcd.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+		first = tmp
+	}
+	return first
+}
+
+// EuclideanGCDRecursive finds GCD of given numbers by recursion
+// first : first number
+// second : second number
+// Recurrance equation : gcd(first,second) =gcd(second,first mod second),if b!=0 gcd(first,second)=first,if b=0
+// Time complexity : O(log min(first,second))
```
Comment (Prakash2403) [inline euclidean_gcd/euclidean_gcd.go]: @yashLadha According to Go documentation, @rashinanwani has used correct convention. So, I think there is no need to use multi-line comments.

Comment (yashLadha) [inline euclidean_gcd/euclidean_gcd.go]: :+1: My bad just refrenced to some other website

**Comment by yashLadha** [comment]
Update the branch, else Looking Good to me. 👍

File: merge_sort/merge_sort.go [hunk 8]
```diff
@@ -0,0 +1,43 @@
+package main
```
Comment (aviaryan) [inline merge_sort/merge_sort.go]: Please remove this file.

File: README.md [hunk 9]
```diff
@@ -93,4 +93,4 @@ Idea by [@Monal5031](https://github.com/Monal5031)
 * [Avi Aryan](https://github.com/aviaryan)
 * [Pratyush Singh](https://github.com/singhpratyush)
 
-Only project maintainers should merge a PR. Other members can add their reviews to a PR but the merging should be done by only a project maintainer.
+Only project maintainers should merge a PR. Other members can add their reviews to a PR but the merging should be done by only a project maintainer.
```
Comment (aviaryan) [inline README.md]: I don't see what this diff changed. It would be great if you can manage to undo it.

File: euclidean_gcd/euclidean_gcd.go [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+		second = first % second
+		first = tmp
+	}
+	return first
+}
+
+// EuclideanGCDRecursive finds GCD of given numbers by recursion
+// first : first number
+// second : second number
+// Recurrance equation : gcd(first,second) =gcd(second,first mod second),if b!=0 gcd(first,second)=first,if b=0
```
Comment (aviaryan) [inline euclidean_gcd/euclidean_gcd.go]: No need for this. We can already see it in the code.

File: euclidean_gcd/euclidean_gcd.go [hunk 11]
```diff
@@ -0,0 +1,37 @@
+package main
+
+import "fmt"
+
+// EuclideanGCD finds GCD of given numbers by iterative method
+// first : first number
+// second : second number
+// tmp : temporary variable which hold values
```
Comment (aviaryan) [inline euclidean_gcd/euclidean_gcd.go]: No need for defining `tmp` as it is not a parameter and also it's use-case is pretty simple.
